maven-model-builder/src/main/java/org/apache/maven/model/profile/DefaultProfileSelector.java

```diff
@@ -89,6 +126,35 @@ public List<Profile> getActiveProfiles(
         return activeProfiles;
     }
 
+    private Map<String, Profile> earlyInterpolateProfileActivations(
+            Collection<Profile> original, ProfileActivationContext context) {
+
+        Model model = new Model();
+
+        UnaryOperator<Profile> activatableProfile = p -> {
+            Profile result = new Profile();
+            result.setId(p.getId());
+            result.setActivation(p.getActivation());
+            return result;
+        };
+        model.setProfiles(original.stream().map(activatableProfile).collect(Collectors.toList()));
+
+        ModelBuildingRequest mbr = new DefaultModelBuildingRequest()
```

Comment:
guess it would be saner to use the actual loaded model we already have at that time and avoid `asProperties` too since request is already there

can be done in `DefaultModelBuilder` probably

side note: we must somehow align it on `master` and not create a behavior in v3 which is different from v4 so for me ensuring v4 behavior (merge) and impl alignment is a prerequisite to this PR.